src/app/dso-shared/dso-edit-metadata/metadata-field-selector/metadata-field-selector.component.ts

```diff
@@ -112,32 +143,33 @@ export class MetadataFieldSelectorComponent implements OnInit, OnDestroy, AfterV
    * Update the mdFieldOptions$ depending on the query$ fired by querying the server
    */
   ngOnInit(): void {
+    this.input.valueChanges.pipe(
+      debounceTime(this.debounceTime),
+      startWith(''),
+      tap( () => this.currentPage$.next(1) )
```

Comment:
It's not a good idea to "perform work" in a tap. Taps are meant for debugging purposes, and shouldn't affect anything else. The main reason for that is that the code in the tap won't be executed if nothing is subscribed to the pipe in the end. Since you're subscribing immediately underneath this line that's not an issue here, but then there's also no good reason not to move this code to the subscribe as well.

Reply:
@artlowel Hi, I changed this observable in the subscribe function